src/libsystemd/sd-bus/bus-convenience.c

```diff
@@ -714,16 +714,16 @@ _public_ int sd_bus_query_sender_privilege(sd_bus_message *call, int capability)
         strjoina(                                               \
                 "type='signal'",                                \
                 sender ? ",sender='" : "",                      \
-                sender ?: "",                                   \
+                strempty(sender),                               \
                 sender ? "'" : "",                              \
                 path ? ",path='" : "",                          \
-                path ?: "",                                     \
+                strempty(path),                                 \
                 path ? "'" : "",                                \
                 interface ? ",interface='" : "",                \
-                interface ?: "",                                \
+                strempty(interface),                            \
                 interface ? "'" : "",                           \
                 member ? ",member='" : "",                      \
-                member ?: "",                                   \
+                strempty(member),                               \
```

Comment:
Hm, this is strange:

```
systemd-resolved[22486]: ../build/src/libsystemd/sd-bus/bus-message.c:1309:51: runtime error: applying zero offset to null pointer
systemd-resolved[22486]:     #0 0x7f2edad35d9b in message_extend_containers /systemd-meson-build/../build/src/libsystemd/sd-bus/bus-message.c:1309:51
systemd-resolved[22486]:     #1 0x7f2edad28917 in message_extend_body /systemd-meson-build/../build/src/libsystemd/sd-bus/bus-message.c:1407:9
systemd-resolved[22486]:     #2 0x7f2edad20f99 in message_append_basic /systemd-meson-build/../build/src/libsystemd/sd-bus/bus-message.c:1601:21
systemd-resolved[22486]:     #3 0x7f2edad2758a in sd_bus_message_append_basic /systemd-meson-build/../build/src/libsystemd/sd-bus/bus-message.c:1637:16
systemd-resolved[22486]:     #4 0x7f2edad318a0 in sd_bus_message_appendv /systemd-meson-build/../build/src/libsystemd/sd-bus/bus-message.c:2428:29
systemd-resolved[22486]:     #5 0x7f2edace1e85 in sd_bus_call_method_asyncv /systemd-meson-build/../build/src/libsystemd/sd-bus/bus-convenience.c:86:21
systemd-resolved[22486]:     #6 0x7f2edace21e9 in sd_bus_call_method_async /systemd-meson-build/../build/src/libsystemd/sd-bus/bus-convenience.c:109:13
systemd-resolved[22486]:     #7 0x7f2edacda605 in sd_bus_request_name_async /systemd-meson-build/../build/src/libsystemd/sd-bus/bus-control.c:203:16
systemd-resolved[22486]:     #8 0x563491ed8d7b in manager_connect_bus /systemd-meson-build/../build/src/resolve/resolved-bus.c:2124:13
systemd-resolved[22486]:     #9 0x563491eba55a in manager_new /systemd-meson-build/../build/src/resolve/resolved-manager.c:642:13
systemd-resolved[22486]:     #10 0x563491eb9680 in run /systemd-meson-build/../build/src/resolve/resolved.c:64:13
systemd-resolved[22486]:     #11 0x563491eb9154 in main /systemd-meson-build/../build/src/resolve/resolved.c:91:1
systemd-resolved[22486]:     #12 0x7f2ed9f07022 in __libc_start_main (/usr/lib/libc.so.6+0x27022)
systemd-resolved[22486]:     #13 0x563491eb899d in _start (/systemd-meson-build/systemd-resolved+0x12299d)
systemd-resolved[22486]: SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior ../build/src/libsystemd/sd-bus/bus-message.c:1309:51 in
```

judging by the location it looks like it's caused by this chunk, but I don't grok why, since `strempty(x)` is just inline wrapper around `x ?: ""`.

Reply:
FWIW, I think the earlier code was easier to read. It's clearer when the same ternary pattern is used explicitly in neighbouring arguments, rather than replace *some* of them with a wrapper.

Reply:
Agreed, reverted (along with with a pretty disgusting Coccinelle patch, so I won't submit this again, in the future).

The fail is still interesting though, wonder if it's a real issue or just some confusion on UBSan's side.

Reply:
Hah, interesting, the issue is still there...

Reply:
Ah, nevermind, the LLVM stack got updated today along with UBSan, so it's something new, albeit unrelated to this PR.